packages/flame/lib/src/game/base_game.dart

```diff
@@ -24,33 +23,20 @@ import 'viewport.dart';
 /// This is the recommended structure to use for most games.
 /// It is based on the Component system.
 class BaseGame extends Game {
+  BaseGame() {
+    components = createComponentSet();
+    _cameraComponent = CameraComponent(Camera(), components);
+  }
+
   /// The list of components to be updated and rendered by the base game.
-  late final ComponentSet components = createComponentSet();
+  late final ComponentSet components;
 
   /// The camera translates the coordinate space after the viewport is applied.
-  final Camera camera = Camera();
+  Camera get camera => _cameraComponent.camera;
 
-  /// The viewport transforms the coordinate space depending on your chosen
-  /// implementation.
-  /// The default implementation no-ops, but you can use this to have a fixed
-  /// screen ratio for example.
-  Viewport get viewport => _viewport;
-
-  Viewport _viewport = DefaultViewport();
-  set viewport(Viewport value) {
-    if (hasLayout) {
-      final previousSize = canvasSize;
-      _viewport = value;
-      onResize(previousSize);
-    } else {
-      _viewport = value;
-    }
-    _combinedProjector = Projector.compose([camera, value]);
-  }
-
-  late Projector _combinedProjector;
-
-  final Vector2 _sizeBuffer = Vector2.zero();
+  // When the Game becomes a Component (#906), this could be added directly
+  // into the component tree.
+  late final CameraComponent _cameraComponent;
```

Comment:
Why can it not be added directly in to the component tree already?

Reply:
I was hitting a snag here: the `onGameResize` method passes `canvasSize` to the camera, but `size` to all other elements. I'm not exactly sure how to approach this issue, but here's what I've been thinking so far:

Currently, BaseGame passes `size = viewport.effectiveSize/camera.zoom` to every component. Essentially, this is the size of the game world visible through the viewport. There is a problem, however: if the camera's zoom level changes, or if the viewport's effective size changes for whatever reason -- the `onGameResize` wouldn't be called. I believe this would be a potential source of bugs for any component that relies on `onGameResize`.

On the other hand, the only component in Flame that currently uses `onGameResize` is `ParallaxComponent`, and even it ignores the passed `size` parameter. So, maybe instead of having `onGameResize` as a lifecycle method, a component would instead subscribe to resize events on either the Game itself or on the camera's viewport?

Honestly, I'm not sure what the correct solution should be. Obviously, the main camera needs to be informed of the canvas size in order to size and position its viewport correctly. On the other hand, components that are "within" the camera do not care about the canvas size -- for them, the viewport is the equivalent of the canvas. For example, if someone wants to have a camera within the camera - the internal camera will only care about the dimensions made available to it by the external camera. So, perhaps we could make each camera responsible for calling `onResize` on all the components that it observes, passing along its viewport size?

Reply:
Right, makes sense. We'll have to think about how to solve that.

> So, perhaps we could make each camera responsible for calling onResize on all the components that it observes, passing along its viewport size?

How would that work if a component is observed by multiple cameras?

Reply:
> How would that work if a component is observed by multiple cameras?

You're absolutely right, this is quite puzzling.
First, it should probably be noted that the vast majority of self-respecting components do not care about the resize events. This is because a "normal" world continues its existence without any regard to whether it's observed through the camera or not. 

There could be interesting examples when you'd want to know whether an object is actively observed by the user or not -- either from an optimization standpoint or to implement some peculiar game mechanic. However, in this case, you'd be interested not only in viewport changes but also in any changes to the camera's target as well.

Viewport changes are obviously of interest to the HUD elements of the camera, but those elements won't be observed by another camera, so it's safe to call onResize for them.

The `ParallaxComponent` is an interesting use-case because, while being a part of the game world, it actually cares about how it is observed. But then, if the same parallax is being observed from two different cameras -- what do we do? I'm not sure what the right solution to parallax is. Maybe instead of relying on the resize event, it could always go into the camera and fetch the current world window size?

Reply:
Maybe we should wait with having the camera as a component, until it can actually be used as a component?
But then just have your restructured camera directly on `Game`?

Reply:
Ok, I made the `CameraComponent` to be a non-component. Now it can't be mounted into the component tree, but at the same time, it will be quick to restore its status as a `Component` when it has more fleshed-out functionality.

Reply:
Great!